docs/mine/lotus/dagstore.md

```diff
@@ -0,0 +1,193 @@
+
+```
+ dagstore
+     |___ index                         # (1)
+     |___ transients                    # (2)
+     |___ datastore                     # (3)
+     |___ .shard-registration-complete  # (4)
+```
+
+1. `index`: holds the shard indices.
+2. `transients`: holds temporary shard data (unsealed pieces) while they're being indexed.
+3. `datastore`: records shard state and metadata so it can survive restarts.
+4. `.shard-registration-complete`: marker file that signals that initial migration is complete.
+
+## First-time migration
+
+When you first start your lotus-miner or the market nodes if you’ve split it out using the lotus that introduced dagstore(we will add the exact version later), a migration process will register all shards in **lazy initialization** mode. As deals come in, shards are fetched and initialized just in time in order to serve the retrieval.
+
+You can monitor the progress of the migration in your log output, by grepping for the keyword `migrator`. Here's example output. Notice the first line, which specifies how many deals will be evaluated (this number includes failed deals that never went on chain, and therefore will not be migrated), and the last lines (which communicate that migration completed successfully):
+
+```
+2021-08-09T22:06:35.701+0300    INFO    dagstore.migrator       dagstore/wrapper.go:286 registering shards for all active deals in sealing subsystem    {"count": 453}
+2021-08-09T22:06:35.701+0300    WARN    dagstore.migrator       dagstore/wrapper.go:335 deal has nil piece CID; skipping        {"deal_id": 0}
+2021-08-09T22:06:35.701+0300    INFO    dagstore.migrator       dagstore/wrapper.go:348 registering deal in dagstore with lazy init     {"deal_id": 2208881, "piece_cid": "baga6ea4seaqhnvxy55e
+nveknyqhkkh7mltcrrcx35yvuxdmcbfouaafkvp6niay"}
+2021-08-09T22:06:35.702+0300    INFO    dagstore.migrator       dagstore/wrapper.go:318 async shard registration completed successfully {"shard_key": "baga6ea4seaqhnvxy55enveknyqhkkh7mltcrrcx
+35yvuxdmcbfouaafkvp6niay"}
+[...]
+2021-08-09T22:06:35.709+0300    INFO    dagstore.migrator       dagstore/wrapper.go:361 finished registering all shards {"total": 44}
+[...]
+2021-08-09T22:06:35.826+0300    INFO    dagstore.migrator       dagstore/wrapper.go:365 confirmed registration of all shards
+2021-08-09T22:06:35.826+0300    INFO    dagstore.migrator       dagstore/wrapper.go:372 successfully marked migration as complete
+2021-08-09T22:06:35.826+0300    INFO    dagstore.migrator       dagstore/wrapper.go:375 dagstore migration complete
+```
+
+## Forcing bulk initialization
+
+Forcing bulk initialization will become important in the near future, when miners begin publishing indices to the network to advertise content they have, and new retrieval features become available (e.g. automatic shard routing).
+
+You should start now if possible, as this process is better carried out gradually and over a longer timeframe, if you have many storage deals.
+
```

Comment:
```suggestion
::: warning
Initialization places IO workload on your storage system. You can stop/start this command at your wish/convenience as proving deadlines approach and elapse, to avoid IOPS starvation or competition with window PoSt.

To stop a bulk initialization(see the next paragraph), press Control-C. Shards being initialized at that time will continue in the background, but no more initializations will be performed. The next time you run the command, it will resume from where it left off.
:::
```